at_client/lib/src/client/at_client_impl.dart

```diff
@@ -311,45 +336,74 @@ class AtClientImpl implements AtClient {
   Future<bool> put(AtKey atKey, dynamic value,
       {bool isDedicated = false}) async {
     AtResponse atResponse = AtResponse();
-    // If the value is neither String nor List<int> throw exception
-    if (value is! String && value is! List<int>) {
-      // TODO: Throw AtValueException
-      throw AtClientException('AT014',
-          'Invalid value type found ${value.runtimeType}. Expected String or List<int>');
-    }
-    if (value is String) {
-      atResponse = await putText(atKey, value);
-    }
-    if (value is List<int>) {
-      atResponse = await putBinary(atKey, value);
+    try {
+      // If the value is neither String nor List<int> throw exception
+      if (value is! String && value is! List<int>) {
+        throw AtValueException(
+            'Invalid value type found ${value.runtimeType}. Expected String or List<int>')
+          ..contextParams = (ContextParams()
+            ..exceptionScenario = ExceptionScenario.invalidValueProvided);
+      }
+      if (value is String) {
+        atResponse = await putText(atKey, value);
+      }
+      if (value is List<int>) {
+        atResponse = await putBinary(atKey, value);
+      }
+    } on AtException catch (e) {
+      e.contextParams ??= ContextParams();
```

Comment:
lines of code inside the catch block can be moved to private method to avoid code duplication.